### PR TITLE
Feature #23 Remove regex phone nr

### PR DESCRIPTION
**Pull request by ewynx**
Issue #23

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -9,5 +9,9 @@ def index
     @countries_collection.delete(["country_name", :COUNTRY_ISO_CODE])
     @countries_collection = @countries_collection.sort_by {|c| c[0]}
     @volunteer_crews = Volunteer.crews.collect{|k,v| [k.humanize, k]}
+
+    # Pre-selected New York
+    @states_collection = CS.states(:US).map { |c| [I18n.transliterate(c[1]), c[0]] }
+    @cities_collection = CS.cities(:NY, :US)
```
Comment (kopchickm) [inline app/controllers/home_controller.rb:15]: Was this done elsewhere?

Comment (house92) [reply app/controllers/home_controller.rb:15]: This actually is in master already, but it seems like sometimes Github gets confused about the order of commits when it creates these comparisons - I've had it happen a few times

Comment (house92) [reply app/controllers/home_controller.rb:15]: The actual diff: https://github.com/panvol/pandemic-volunteers/compare/master...ewynx:remove-regex-phone-nr

**Comment by house92** [comment]
> We wanted to remove the regex requirement for the phones since the regex I wrote didn't cover some generated phone numbers during testing.
> 
> We could discuss phone requirements on the issue: #23
> 
> On the other hand, this PR also remove phone number requirements from the form validation. But phone is still needed in the DB as I think it should be.

The form validation may be faulty. There can be local numbers which are shorter than 7 digits, so all we can really do is assert that any number has been entered, which is as good as pointless if a user can simply enter "0" to bypass the validation. I'd suggest that this form validation be removed, as has been done in this PR, and we look at a solution like https://numverify.com/documentation if we really want to verify numbers properly

**Comment by kopchickm** [comment]
I agree with @house92 about dropping the validation entirely. I'd rather be overly permissive.

**Comment by hantuzun** [comment]
Hi @house92.

I agree with you, let's drop the validation. Do you think that should we "require" phone number?

I think if somebody doesn't dare to share their number, they'd be less likely to actually do volunteering.

BTW, thanks for sharing https://numverify.com/documentation -- good to know.

**Comment by hantuzun** [comment]
This has been handled in another PR instead: https://github.com/panvol/pandemic-volunteers/pull/39

**Pull request closed without merging**